hair_salon/data/ir_actions_server.xml

```diff
@@ -0,0 +1,61 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo>
+    <record id="break_booking_server_action" model="ir.actions.server">
+        <field name="code"><![CDATA[
+env['calendar.event'].search([('x_parent_id', '=', record.id)]).unlink()
+        ]]></field>
+        <field name="model_id" ref="calendar.model_calendar_event"/>
+        <field name="state">code</field>
+        <field name="name">Execute Code</field>
+    </record>
+    <record id="delete_subpart_main_calendar_event_server_action" model="ir.actions.server">
+        <field name="child_ids" eval="[(6, 0, [ref('break_booking_server_action')])]"/>
+        <field name="code"><![CDATA[parent_record = env['calendar.event'].browse(record.x_parent_id)
+parent_record.x_appointment_part_2.unlink()
```

Comment:
waht about part_1?

Reply:
You can only delete first part tot delete both, otherwise I cannot find a conditions.